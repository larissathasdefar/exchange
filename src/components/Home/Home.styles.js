import styled from 'styled-components'
import Paper from '@material-ui/core/Paper'

export const Container = styled(Paper)`
  max-width: 1000px;
  margin: 0 auto;
  padding: 20px 14px;
`

export const Content = styled.div``

export const ActionsContainer = styled.div`
  display: flex;
  justify-content: space-around;
  align-items: center;
  padding: 8px;
`
