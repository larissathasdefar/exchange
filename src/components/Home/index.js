import React, { useState } from 'react'
import _ from 'prop-types'
import Button from '@material-ui/core/Button'
import Typography from '@material-ui/core/Typography'
import Stepper from '../UI/Stepper'
import { Container, Content, ActionsContainer } from './Home.styles'

const currencies = [
  { name: 'American Dollar', symbol: '$', code: 'USD' },
  { name: 'British Pound', symbol: '£', code: 'GPB' },
  { name: 'Euro', symbol: '€', code: 'EUR' },
]

const Home = ({ history }) => {
  const [currency, setCurrency] = useState('GPB')
  return (
    <Container elevation={1}>
      <Content>
        <Typography variant="h4" paragraph>Wallet</Typography>
        <Stepper
          steps={[
            { title: '£48.54', subtitle: 'GPB - British Pound', code: 'GPB' },
            { title: '€138.40', subtitle: 'EUR - Euro', code: 'EUR' },
            { title: '$60.00', subtitle: 'USD - American Dollar', code: 'USD' },
          ]}
          align="center"
          onChangeStep={(index, step) => setCurrency(step.code)}
        />
      </Content>
      <Typography paragraph variant="h5">
        Exchange with:
      </Typography>
      <ActionsContainer>
        {
          currencies.map(({ name, symbol, code }) => (
            <Button
              key={code}
              variant="outlined"
              disabled={currency === code}
              onClick={() => history.push(`/exchange/${currency}/to/${code}`)}>
              {`${symbol} - ${name}`}
            </Button>
          ))
        }
      </ActionsContainer>
    </Container>
  )
}

Home.propTypes = {
  history: _.shape({
    push: _.func.isRequired,
  }).isRequired,
}

export default Home
