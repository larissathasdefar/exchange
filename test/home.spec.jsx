import React from 'react'
import { shallow } from 'enzyme'

import Home from 'components/Home'

describe('Home', () => {
  it('Having 2 pockets', () => {
    const component = shallow(
      <Home
        user={{
          pockets: [
            { code: 'USD', amount: 44.90 },
            { code: 'GBP', amount: 50.56 },
          ],
          photo: 'https://scontent.fjoi1-2.fna.fbcdn.net/v/t1.0-1/c21.0.699.699a/s100x100/13240613_1031098473634742_623670164829509408_n.jpg?_nc_cat=106&_nc_ht=scontent.fjoi1-2.fna&oh=73ef6448c5749f90458e20e6bd8afe64&oe=5CEC15C9',
          name: 'Larissa Farias',
        }}
        currencies={{
          USD: 'United States Dollar',
          GBP: 'British Pound Sterling',
          EUR: 'Euro',
        }}
        history={{ push: route => console.log(route) }}
      />
    )
    expect(component.debug()).toMatchSnapshot()
  })
})