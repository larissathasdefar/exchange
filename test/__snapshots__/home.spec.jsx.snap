// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Having 2 pockets 1`] = `
"<Styled(WithStyles(Paper)) elevation={1}>
  <styled.div>
    <Styled(WithStyles(Typography)) variant=\\"h4\\" paragraph={true}>
      Pockets
    </Styled(WithStyles(Typography))>
    <CustomStepper steps={{...}} align=\\"center\\" textStyle={{...}} iconStyle={{...}} start={0} />
  </styled.div>
  <styled.div>
    <WithStyles(Button) variant=\\"outlined\\" onClick={[Function: onClick]}>
      Exchange
    </WithStyles(Button)>
  </styled.div>
</Styled(WithStyles(Paper))>"
`;
